### PR TITLE
fix(cli): resolve commands on Windows with one shell string so Node 24 stops printing DEP0190

### LINKED ISSUE #1394: bug(cli): Windows PATH lookup prints DEP0190 on Node 24 (shell true with an args array)
## What happens

On Windows with Node 24, `egc init` (and any command that resolves a tool on PATH) prints:

```
(node:13256) [DEP0190] DeprecationWarning: Passing args to a child process with shell option true can lead to security vulnerabilities, as the arguments are not escaped, only concatenated.
```

Reported three times by @Akisolu, on v1.1.17 (#1049), v1.1.18 (#1217) and v1.1.21 (#1380), always on Windows 10 with Node 24. Harmless today, but Node has announced the pattern will become an error, and the warning sits in the middle of the install output on the platform where people read it most carefully.

## Where

`scripts/lib/utils.js`, the Windows branch of the PATH lookup:

```js
const result = spawnSync('where', [cmd], { stdio: 'pipe', shell: true });
```

`shell: true` is there on purpose (it inherits `PATHEXT`, so `where npm` finds `npm.cmd`), but Node 24 deprecates passing an args array together with it.

## Expected

Keep the shell (the `PATHEXT` behaviour is the point) and pass one string instead of an array. `cmd` is already validated against `^[a-zA-Z0-9_.-]+$` a few lines above, so concatenation is safe:

```js
const result = spawnSync(`where ${cmd}`, { stdio: 'pipe', shell: true });
```

A test can spawn a small Node script that calls the lookup on `win32` and assert that stderr carries no `DEP0190` line. `scripts/crush-run.js` already joins its arguments into one string for the same reason; the same shape applies here.

### PR DESCRIPTION
**Pull request by Fmarzochi**
## What Changed

`scripts/lib/utils.js`, `commandExists`: the Windows branch calls `where` through the shell with one string (`where <cmd>`) instead of an args array. The shell stays, because it is what makes `where npm` resolve `npm.cmd` through `PATHEXT`; the command name is validated to letters, digits, dot, dash and underscore before this point, so joining it needs no escaping.

`tests/lib/utils.test.js`: a new case runs `commandExists('node')` in a child process and asserts exit 0, `true` on stdout, and no `DEP0190` on stderr. The warning is printed by the runtime once per process, which is why the child process is needed to observe it; on Windows it exercises the branch that used to warn, elsewhere it guards the POSIX branch.

## Why This Change

Node 24 deprecates passing an args array together with `shell: true` and prints `[DEP0190] DeprecationWarning` on every `egc init` on Windows. It has been in every Windows report since v1.1.17 without a ticket. Fixes #1394.

## Testing Done

- [x] Manual testing completed
- [x] Automated tests pass locally (`node tests/run-all.js`)
- [x] Edge cases considered and tested
- [ ] If this PR touches a file shared across concurrent EGC processes (encryption key, state files, install-state, lockfiles under `~/.egc/`), a concurrent-access test was added or updated (not applicable)

## Type of Change
- [x] `fix:` Bug fix

## Security & Quality Checklist
- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [x] Shell scripts pass shellcheck (if applicable)
- [x] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation
- [x] Updated relevant documentation (comment at the call site)
- [x] Added comments for complex logic
- [ ] README updated (not needed)

## Credit

Reported by @Akisolu on Windows 10 with Node 24 in #1049, #1217 and #1380.

---
## Summary by cubic
Stops Node 24 from printing a DEP0190 deprecation warning on Windows during `egc init` by passing `where` a single shell string instead of an args array, preserving `PATHEXT` resolution. The command name is validated to safe characters before this point, so no escaping is needed. Fixes #1394.

- Adds a child-process test asserting `commandExists('node')` finds the command and prints no DEP0190 on stderr, using the shared subprocess timeout fixture.

<sup>Written for commit dd46dcb9f227f62db4f3a992359cb7a3b85b2f27. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Fmarzochi/EGC/pull/1397?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: tests/lib/utils.test.js [hunk 1]
```diff
@@ -586,6 +587,29 @@ function runTests() {
...
+    // process, on stderr, by the runtime itself.
+    const utilsPath = path.join(__dirname, '..', '..', 'scripts', 'lib', 'utils.js');
+    const scriptDir = fs.mkdtempSync(path.join(os.tmpdir(), 'egc-command-exists-'));
+    const scriptPath = path.join(scriptDir, 'probe.js');
+    try {
+      fs.writeFileSync(scriptPath, [
+        `const utils = require(${JSON.stringify(utilsPath)});`,
+        "process.stdout.write(String(utils.commandExists('node')));",
+      ].join('\n'));
+      const result = spawnSync(process.execPath, [scriptPath], { encoding: 'utf8', timeout: 10000 });
```
[the remark this answers is not in the document]
Comment (Fmarzochi) [reply tests/lib/utils.test.js]: Taken in dd46dcb9: the probe takes CLI_TIMEOUT_MS from tests/fixtures/subprocess-timeouts.js.
